modules/auxiliary/scanner/smb/smb2.rb

```diff
@@ -12,6 +12,9 @@ class MetasploitModule < Msf::Auxiliary
   include Msf::Auxiliary::Scanner
   include Msf::Auxiliary::Report
 
+  include Msf::Module::Deprecated
+  deprecated(Date.new(2020, 10, 6))
```

Comment:
What are your thoughts on adding an affordance to users about why it's deprecated?

```suggestion
  deprecated(Date.new(2020, 10, 6), "why it's being deprecated / what to do instead")
```

Reply:
I had to cherry-pick the commit in which you added this functionality over, but I got it done and now these two modules have meaningful reason messages.

```
msf5 auxiliary(scanner/smb/smb_version) > use auxiliary/scanner/smb/smb1 

[!] *                  The module auxiliary/scanner/smb/smb1 is deprecated!                  *
[!] *                   This module will be removed on or about 2020-10-06                   *
[!] *            Use auxiliary/scanner/smb/smb_version for SMB version detection             *
msf5 auxiliary(scanner/smb/smb1) >
```